### PR TITLE
#19 Use RecurringType from PaymentProvider

### PR DESCRIPTION
**Pull request by BLangendorf**
When Recurring is enabled, the `RecurringType` is in the `PaymentProvider` and is now used to set the `RecurringContract` in `PaymentRequest`s.

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/PaymentRequestBuilder.java [hunk 1]
```diff
@@ -145,17 +145,17 @@ public PaymentRequestBuilder withShopperName(final Name shopperName) {
         return this;
     }
 
-    public PaymentRequestBuilder withRecurringContractForUser(final UserData userData) {
+    public PaymentRequestBuilder withRecurringContractForUser() {
         if (paymentInfo.getPaymentProvider().isRecurringEnabled()) {
-            final Recurring recurring = createRecurring(RECURRING_CONTRACT);
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/PaymentRequestBuilder.java]: Can we get rid of the constant now?

**Comment by pierre** [comment]
Could you also document in the README how to configure it using plugin properties (`recurringType`) and the possible values? @arnelandwehr summarized the behavior here: https://github.com/killbill/killbill-adyen-plugin/pull/14#discussion_r41135172

File: src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/PaymentRequestBuilder.java (hunk 1 above)
Comment (BLangendorf) [inline src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/PaymentRequestBuilder.java]: Commit 84547e6d85877bc232f01a87173f0869750d8cbf

**Comment by BLangendorf** [comment]
Commit c5db47cb0ce1a3be6c7bbd7b3d0bf2db98c22672

File: src/test/java/org/killbill/billing/plugin/adyen/client/payment/builder/BaseTestPaymentRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/adyen/client/payment/builder/BaseTestPaymentRequestBuilder.java]: Missing copyright header.

**Comment by pierre** [comment]
:tada:

**Pull request merged**